web/source/osk/preProcessor.ts

```diff
@@ -47,8 +47,14 @@ namespace com.keyman.osk {
         // Deadkey matching continues to be troublesome.
         // Deleting matched deadkeys here seems to correct some of the issues.   (JD 6/6/14)
         outputTarget.deadkeys().deleteMatched();      // Delete any matched deadkeys before continuing
-  
-        let Lkc = PreProcessor._GetClickEventProperties(e['key'].spec as keyboards.ActiveKey, Lelem);
+
+        let keySpec = (e['key'] ? e['key'].spec : null) as keyboards.ActiveKey;
+        if(!keySpec) {
+          console.error("OSK key with ID '" + e.id + "', keyID '" + e.keyId + "' missing needed specification");
```

Comment:
How could this happen? Can you give an example so we can avoid producing keyboards that trigger this issue?

Reply:
Your guess is as good as mine; that's exactly why I added the console log.

My first thought is that _maybe_ a spacing element is somehow "getting clicked". OSK elements not intended for input (internal spacing, formatting, etc) don't usually have the `'key'` property set, after all - only key elements that _are_ intended for input get that.

That being said, I can only guess... so I figured I'd add logging that should hopefully make the underlying issue more explicit.  I don't think I've ever seen the bug during personal tests, but it obviously can occur... somehow.